src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,180 @@
+    }
+
+    @Test
+    void register_shortLogin_NotOk() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(invalidLogin);
+        john.setPassword(validPassword);
+        george.setAge(ANOTHER_ADULT_AGE);
+        george.setLogin(shortLogin);
+        george.setPassword(anotherValidPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+        assertThrows(InvalidDataException.class, () -> registrationService.register(george));
+    }
+
+    @Test
+    void register_shortPassword_NotOk() {
+        john.setAge(ANOTHER_ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(invalidPassword);
+        george.setAge(ADULT_AGE);
+        george.setLogin(thirdValidLogin);
+        george.setPassword(shortPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+        assertThrows(InvalidDataException.class, () -> registrationService.register(george));
+    }
+
+    @Test
+    void register_emptyLogin_NotOk() {
+        john.setAge(ADULT_AGE);
+        john.setLogin("");
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void register_emptyPassword_NotOK() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword("");
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void register_nullUser_NotOk() {
+        assertThrows(InvalidDataException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_nullLogin_NotOk() {
+        john.setAge(ANOTHER_ADULT_AGE);
+        john.setLogin(null);
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void register_nullPassword_NotOk() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(null);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void register_duplicateUser_NotOk() {
+        john.setAge(ANOTHER_ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(validPassword);
+        Storage.people.add(john);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void register_userWithExistingLogin_NotOk() {
+        john.setAge(ANOTHER_ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(validPassword);
+        Storage.people.add(john);
+        maria.setAge(ANOTHER_ADULT_AGE);
+        maria.setLogin(validLogin);
+        maria.setPassword(anotherValidPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(maria));
```

Comment:
The test for registering a user with an existing login is good but ensure that the `register` method has the logic to check for existing logins, not just duplicate user objects.